src/sync/app.cpp

```diff
@@ -586,10 +586,9 @@ void App::log_in_with_credentials(const AppCredentials& credentials,
     }, handler);
 }
 
-void App::log_out(std::function<void (Optional<AppError>)> completion_block) const
+void App::log_out(std::shared_ptr<SyncUser> user, std::function<void (Optional<AppError>)> completion_block) const
```

Comment:
Should this be user_id instead? For easy compatibility.

Reply:
I don't have a strong opinion, either works for me. I'll change it.

Reply:
Or actually, no. 

There are some checks on the user inside `log_out`, so parsing the identity would require looking up the user again? That seems cumbersome and I don't think there are any cases where you don't have a reference to the shared pointer for the user for this?

Reply:
Alright, that's fine.